File: backend/src/domain/enums.py
# ============================================================================
# Enums - Domain Enumerations
# ============================================================================

"""
Domain enumerations for the Parking Management System.
"""

from enum import Enum, auto


class VehicleType(str, Enum):
    """Types of vehicles."""
    CAR = "car"
    TRUCK = "truck"
    MOTORCYCLE = "motorcycle"
    BUS = "bus"
    BICYCLE = "bicycle"


class PowerSource(str, Enum):
    """Power source for vehicles."""
    GASOLINE = "gasoline"
    DIESEL = "diesel"
    ELECTRIC = "electric"
    HYBRID = "hybrid"


class SlotType(str, Enum):
    """Types of parking slots."""
    REGULAR = "regular"
    EV = "ev"
    DISABLED = "disabled"
    COMPACT = "compact"
    LARGE = "large"


class SlotStatus(str, Enum):
    """Status of a parking slot."""
    AVAILABLE = "available"
    OCCUPIED = "occupied"
    RESERVED = "reserved"
    MAINTENANCE = "maintenance"


class BookingStatus(str, Enum):
    """Status of a parking booking."""
    PENDING = "pending"
    CONFIRMED = "confirmed"
    ACTIVE = "active"
    COMPLETED = "completed"
    CANCELLED = "cancelled"
    EXPIRED = "expired"


class ChargingStatus(str, Enum):
    """Status of EV charging."""
    IDLE = "idle"
    CHARGING = "charging"
    COMPLETED = "completed"
    ERROR = "error"
    PAUSED = "paused"


class PaymentStatus(str, Enum):
    """Status of payment."""
    PENDING = "pending"
    PROCESSING = "processing"
    COMPLETED = "completed"
    FAILED = "failed"
    REFUNDED = "refunded"
    CANCELLED = "cancelled"


class NotificationType(str, Enum):
    """Types of notifications."""
    BOOKING_CONFIRMED = "booking_confirmed"
    BOOKING_REMINDER = "booking_reminder"
    PARKING_ALERT = "parking_alert"
    CHARGING_STARTED = "charging_started"
    CHARGING_COMPLETED = "charging_completed"
    PAYMENT_SUCCESS = "payment_success"
    PAYMENT_FAILED = "payment_failed"
    SYSTEM = "system"

File: shared/types/src/enums.ts
// ============================================================================
// Enums - Shared Enum Definitions
// ============================================================================

// parking-management-system/shared/types/src/enums.ts

/**
 * Booking status enum
 */
export enum BookingStatus {
  PENDING = 'pending',
  CONFIRMED = 'confirmed',
  ACTIVE = 'active',
  COMPLETED = 'completed',
  CANCELLED = 'cancelled',
  NO_SHOW = 'no_show',
}

/**
 * Payment status enum
 */
export enum PaymentStatus {
  PENDING = 'pending',
  PROCESSING = 'processing',
  COMPLETED = 'completed',
  FAILED = 'failed',
  REFUNDED = 'refunded',
  CANCELLED = 'cancelled',
}

/**
 * Payment method enum
 */
export enum PaymentMethod {
  CARD = 'card',
  PAYPAL = 'paypal',
  APPLE_PAY = 'apple_pay',
  GOOGLE_PAY = 'google_pay',
  BANK = 'bank',
  CRYPTO = 'crypto',
}

/**
 * Notification type enum
 */
export enum NotificationType {
  BOOKING = 'booking',
  PAYMENT = 'payment',
  PARKING = 'parking',
  CHARGING = 'charging',
  SYSTEM = 'system',
  PROMOTION = 'promotion',
  REMINDER = 'reminder',
}

/**
 * Notification priority enum
 */
export enum NotificationPriority {
  HIGH = 'high',
  NORMAL = 'normal',
  LOW = 'low',
}

/**
 * Charging status enum
 */
export enum ChargingStatus {
  AVAILABLE = 'available',
  OCCUPIED = 'occupied',
  RESERVED = 'reserved',
  CHARGING = 'charging',
  PAUSED = 'paused',
  COMPLETED = 'completed',
  ERROR = 'error',
  MAINTENANCE = 'maintenance',
}

/**
 * Connector type enum
 */
export enum ConnectorType {
  TYPE1 = 'type1',
  TYPE2 = 'type2',
  CCS = 'ccs',
  CHADEMO = 'chademo',
  TESLA = 'tesla',
}

/**
 * Charging profile enum
 */
export enum ChargingProfile {
  STANDARD = 'standard',
  FAST = 'fast',
  RAPID = 'rapid',
  ULTRA_RAPID = 'ultra_rapid',
}

/**
 * Vehicle type enum
 */
export enum VehicleType {
  CAR = 'car',
  SUV = 'suv',
  TRUCK = 'truck',
  MOTORCYCLE = 'motorcycle',
  BICYCLE = 'bicycle',
  VAN = 'van',
}

/**
 * Environment enum
 */
export enum Environment {
  DEVELOPMENT = 'development',
  STAGING = 'staging',
  PRODUCTION = 'production',
  TEST = 'test',
}

/**
 * Language enum
 */
export enum Language {
  EN = 'en',
  ES = 'es',
  FR = 'fr',
  DE = 'de',
  IT = 'it',
  PT = 'pt',
  RU = 'ru',
  ZH = 'zh',
  JA = 'ja',
  KO = 'ko',
}

/**
 * Currency enum
 */
export enum Currency {
  USD = 'USD',
  EUR = 'EUR',
  GBP = 'GBP',
  JPY = 'JPY',
  CNY = 'CNY',
  INR = 'INR',
  CAD = 'CAD',
  AUD = 'AUD',
  BRL = 'BRL',
}